backend/reservations/services/guest_ingestion_service.py

```diff
@@ -0,0 +1,238 @@
+        from ..helpers import ingest_csv
+        from pathlib import Path
+
+        file_path = config.get('file_path')
+        if not file_path or not Path(file_path).exists():
+            raise ValueError(f"CSV file not found: {file_path}")
+
+        fields, guests = ingest_csv(file_path)
+
+        self.logger.info(f"Loaded {len(guests)} clean records from CSV with fields: {fields}")
+
+        return {
+            'guests': guests,
+            'fields': fields,
+            'metadata': {
+                'source_file': file_path,
+                'total_records': len(guests),
+                'fields_found': fields
+            }
+        }
+
+    def _transform_external_data(self, raw_data: Dict[str, Any],
+                               source_name: str) -> Dict[str, Any]:
+        if source_name.lower() == 'secretparty':
+            return self._transform_secretparty_data(raw_data)
+        elif source_name.lower() == 'csv':
+            return self._transform_csv_data(raw_data)
+        elif source_name.lower() == 'manual':
+            return raw_data  # Assume already in correct format
+        else:
+            raise ValueError(f"No transformer for source: {source_name}")
+
+    def _transform_secretparty_data(self, raw_data: Dict[str, Any]) -> Dict[str, Any]:
+        from ..ingest_models import SecretPartyGuestIngest
+
+        self.logger.info(f"Transforming {len(raw_data.get('tickets', []))} SecretParty tickets")
+
+        transformed_guests = []
+        for ticket in raw_data.get('tickets', []):
+            try:
+                # Use the Pydantic model to handle the transformation
+                guest_obj = SecretPartyGuestIngest.from_source(ticket, 'json')
+
+                transformed_guests.append(guest_obj)  # Keep as objects, not dicts
+
+            except Exception as e:
+                self.logger.warning(f"Failed to transform ticket {ticket.get('id', 'unknown')}: {e}")
+                continue
+
+        self.logger.info(f"Successfully transformed {len(transformed_guests)} tickets to guest records")
+
+        return {
+            'guests': transformed_guests,
+            'metadata': raw_data.get('metadata', {}),
+            'source': 'secretparty'
+        }
+
+    def _transform_csv_data(self, raw_data: Dict[str, Any]) -> Dict[str, Any]:
+        from ..ingest_models import SecretPartyGuestIngest
+
+        guests = []
+
+        for row in raw_data.get('guests', []):
+            try:
+                # Use the Pydantic model to handle CSV transformation
+                guest_obj = SecretPartyGuestIngest.from_source(row, 'csv')
+                guests.append(guest_obj)  # Keep as objects, not dicts
+            except Exception as e:
+                self.logger.warning(f"Failed to transform CSV row: {e}")
+                continue
+
+        return {
+            'guests': guests
+        }
+
+    def _process_ingestion_data(self, transformed_data: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Nitpick (non-blocking): This method is quite long. Consider breaking it into several helper methods.